aws.go

```diff
@@ -82,6 +83,33 @@ func (a *AWS) Upload(local, remote string) error {
 		}
 	}
 
+
+
+	/*
+		Set content encoding in s3 while upload.
+		Use: PutObjectInput->ContentEncoding(string)
+		Works only with files containing an extension with a single dot like ".tgz" not ".tar.gz".
+		- usage
+		publish:
+  			s3_sync:
+				content_encoding:
+      				".jgz": gzip
+      				".cgz": gzip
+      				".tgz": gzip
+	*/
+	var contentEncoding string
+	if a.vargs.ContentEncoding.IsString() {
+		contentEncoding = a.vargs.ContentEncoding.String()
+	} else if !a.vargs.ContentEncoding.IsEmpty() {
+		contentMap := a.vargs.ContentEncoding.Map()
+		for patternExt := range contentMap {
+			if patternExt == fileExt {
```

Comment:
we use glob matching everywhere else so it feels counterintuitive to have this only work as a direct extension match. i would vote for making this a glob match also.